src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -270,31 +275,60 @@ def _publish_trade_event(self, trade, is_bid_trade):
             # the bid clearing and one for the offer clearing.
             return
 
-        event_response_dict = {"device_info": self._device_info_dict,
-                               "event": "trade",
-                               "trade_id": trade.id,
-                               "time": trade.time.isoformat(),
-                               "price": trade.offer.price,
-                               "energy": trade.offer.energy,
-                               "fee_price": trade.fee_price,
-                               "area_uuid": self.device.uuid,
-                               "seller": trade.seller
-                               if trade.seller == self.device.name else "anonymous",
-                               "buyer": trade.buyer
-                               if trade.buyer == self.device.name else "anonymous",
-                               "residual_id": trade.residual.id
-                               if trade.residual is not None else "None"}
-
-        bid_offer_key = 'bid_id' if is_bid_trade else 'offer_id'
-        event_response_dict["event_type"] = "buy" \
-            if trade.buyer == self.device.name else "sell"
-        event_response_dict[bid_offer_key] = trade.offer.id
-        trade_event_channel = f"{self.channel_prefix}/events/trade"
-        if self.connected:
-            self.redis.publish_json(trade_event_channel, event_response_dict)
-
         if self.is_aggregator_controlled:
+            event_response_dict = {'event': 'trade',
+                                   'asset_id': self.device.uuid,
```

Comment:
we can build the common keys outside the conditional if, but let's leave it for now

Reply:
Yeah, this is from the time when i thought that the direct connection will be deprecated.